File: app/src/main/java/com/example/torgammelgard/pokerhourly/GraphFragment.java
package com.example.torgammelgard.pokerhourly;

import android.graphics.Color;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;

import com.jjoe64.graphview.GraphView;
import com.jjoe64.graphview.GraphViewSeries;
import com.jjoe64.graphview.GraphViewStyle;
import com.jjoe64.graphview.LineGraphView;

import java.util.ArrayList;

public class GraphFragment extends android.support.v4.app.Fragment {
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.graph_frag, container, false);
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        updateGraph();
    }

    private void updateGraph() {
        /* Prepare the graphView data*/
        ArrayList<Session> sessionList = ((MainApp)getActivity().getApplication())
                .mDataSource.getLastSessions(20);
        int[] resultIntArray = new int[sessionList.size() + 1];
        int index = 0;
        resultIntArray[index++] = 0;
        for (Session session : sessionList) {
            resultIntArray[index++] = session.getResult();
        }
        GraphView.GraphViewData[] data = makeData(resultIntArray);

        /* style - nested class*/
        GraphViewSeries.GraphViewSeriesStyle g =
                new GraphViewSeries.GraphViewSeriesStyle(Color.GREEN, 7);

        /* init the graph */
        GraphViewSeries graphViewSeries = new GraphViewSeries("graphViewSeries", g, data);
        GraphView graphView = new LineGraphView(getActivity(), getString(R.string.graph_title));

        /* Style the graph */
        GraphViewStyle style = new GraphViewStyle();
        style.setNumHorizontalLabels(11);
        graphView.setGraphViewStyle(style);

        /* add the data and show*/
        graphView.addSeries(graphViewSeries);
        if (getView() != null) {
            LinearLayout layout = (LinearLayout) getView().findViewById(R.id.graphLayout);
            layout.addView(graphView);
        }
    }

    /** returns data for the GraphView */
    private GraphView.GraphViewData[] makeData(int[] dataList) {
        double sum = 0;
        GraphView.GraphViewData[] data = new GraphView.GraphViewData[dataList.length];
        for (int index = 0; index < data.length; index++) {
            sum += (double)dataList[index] / 100;
            data[index] = new GraphView.GraphViewData(index, sum);
        }
        return data;
    }
}

File: app/src/main/kotlin/se/torgammelgard/pokertrax/fragments/GraphFragment.kt
package se.torgammelgard.pokertrax.fragments

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.jjoe64.graphview.GraphView
import com.jjoe64.graphview.series.DataPoint
import com.jjoe64.graphview.series.LineGraphSeries
import dagger.android.support.AndroidSupportInjection
import org.jetbrains.anko.doAsync
import org.jetbrains.anko.onComplete
import se.torgammelgard.pokertrax.R
import se.torgammelgard.pokertrax.model.repositories.SessionRepository
import javax.inject.Inject

class GraphFragment : Fragment() {

    @Inject lateinit var sessionRepository: SessionRepository

    override fun onAttach(context: Context?) {
        AndroidSupportInjection.inject(this)
        super.onAttach(context)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.graph_frag, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        updateGraph()
    }

    private fun updateGraph() {
        val graph = view?.findViewById<GraphView>(R.id.graph)

        // get and construct the data
        graph?.let {
            doAsync {
                val sessions = sessionRepository.getAllSessions()
                onComplete {
                    val dataPoints = arrayOfNulls<DataPoint>(sessions.size + 1)

                    // start data at 0, 0
                    dataPoints[0] = DataPoint(0.0, 0.0)

                    // add the rest of the results
                    sessions.forEachIndexed { index, session ->
                        dataPoints[index + 1] = DataPoint(index.toDouble(), (session.result / 100).toDouble())
                    }

                    val series = LineGraphSeries<DataPoint>(dataPoints)

                    // style graph
                    graph.viewport.isXAxisBoundsManual = true
                    graph.viewport.setMinX(0.0)
                    graph.viewport.setMaxX(dataPoints.size - 0.5)

                    graph.addSeries(series)
                }
            }
        }
    }
}